.github/workflows/on-pr.yml

```diff
@@ -0,0 +1,41 @@
+name: "Validate and plan"
+
+on:
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        java-version: [ 8 ]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: ${{ matrix.java-version }}
+          cache: 'maven'
+      - name: Configure git user
+        run: |
+          git config user.email "actions@github.com"
+          git config user.name "GitHub Actions"
+      - name: Build and test
+        run: |
+          mvn clean install -Dmaven.javadoc.skip --quiet
+          mvn jacoco:report jacoco:report-integration
+          mvn package -P all-modules
```

Comment:
Why run `package` again? You already did an `install`, which includes `package`. Strictly speaking, to run tests, you only need to run to the `verify` phase anyway. Also, the `all-modules` profile was removed some time ago.